src/librustc/middle/stability.rs

```diff
@@ -402,7 +402,8 @@ impl<'a, 'tcx> Index<'tcx> {
 
         let mut is_staged_api = false;
         for attr in &krate.attrs {
-            if attr.name() == "stable" || attr.name() == "unstable" {
+            let name = match attr.name() { Some(name) => name, _ => continue };
```

Comment:
worth factoring this idiom out into a macro? Maybe doesn't get used enough, but I'm not really a fan of match one-liners